src/Aws/Common/Credentials/RefreshableInstanceProfileCredentials.php

```diff
@@ -42,6 +42,21 @@ public function __construct(CredentialsInterface $credentials, InstanceMetadataC
         $this->client = $client ?: InstanceMetadataClient::factory();
     }
 
+    public function serialize()
+    {
+        return json_encode(array(
+            'credentials' => parent::serialize(),
+            'client' => serialize($this->client),
```

Comment:
The anonymous functions would not be serialized. The only part of the client that would get serialized would be its config array, and unserializing it would just use that config to call the constructor. 

The other option would be to use a `__wakeup` method instead and have it call `InstanceMetadataClient::factory()`, i.e., mimic what happens when you [create a new RefreshableInstanceProfileCredentials without a client](https://github.com/aws/aws-sdk-php/blob/2.8/src/Aws/Common/Credentials/RefreshableInstanceProfileCredentials.php#L42). (This is what is done by default in [`Credentials::factory()`](https://github.com/aws/aws-sdk-php/blob/2.8/src/Aws/Common/Credentials/RefreshableInstanceProfileCredentials.php#L42).) Doing so would involve fewer changes but would wipe out any custom InstanceMetadataClient, but I'm not sure anyone has ever created or used a custom one of those.